From 063676160e5cd96acf399c9d1da86ef83f539c6e Mon Sep 17 00:00:00 2001
From: codeworkx <daniel.hillenbrand@codeworkx.de>
Date: Mon, 28 Aug 2017 20:54:02 +0200
Subject: [PATCH 1/2] init: panic-reboot to bootloader only on eng builds

Since our regular builds are from type "userdebug" and not "user",
we have to care about this here.

Change-Id: If6c6f8adebf1c59fc8448f61793c2c3debca6899
---
 init/Android.bp | 4 ++--
 init/Android.mk | 8 ++++----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/init/Android.bp b/init/Android.bp
index aaef7e9..c12456a 100644
--- a/init/Android.bp
+++ b/init/Android.bp
@@ -40,8 +40,6 @@ cc_defaults {
                 "-DALLOW_LOCAL_PROP_OVERRIDE=1",
                 "-UALLOW_PERMISSIVE_SELINUX",
                 "-DALLOW_PERMISSIVE_SELINUX=1",
-                "-UREBOOT_BOOTLOADER_ON_PANIC",
-                "-DREBOOT_BOOTLOADER_ON_PANIC=1",
                 "-UWORLD_WRITABLE_KMSG",
                 "-DWORLD_WRITABLE_KMSG=1",
                 "-UDUMP_ON_UMOUNT_FAILURE",
@@ -52,6 +50,8 @@ cc_defaults {
             cppflags: [
                 "-USHUTDOWN_ZERO_TIMEOUT",
                 "-DSHUTDOWN_ZERO_TIMEOUT=1",
+                "-UREBOOT_BOOTLOADER_ON_PANIC",
+                "-DREBOOT_BOOTLOADER_ON_PANIC=1",
             ],
         },
     },
diff --git a/init/Android.mk b/init/Android.mk
index 1ec5a62..8839488 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -8,22 +8,22 @@ ifneq (,$(filter userdebug eng,$(TARGET_BUILD_VARIANT)))
 init_options += \
     -DALLOW_LOCAL_PROP_OVERRIDE=1 \
     -DALLOW_PERMISSIVE_SELINUX=1 \
-    -DREBOOT_BOOTLOADER_ON_PANIC=1 \
     -DDUMP_ON_UMOUNT_FAILURE=1
 else
 init_options += \
     -DALLOW_LOCAL_PROP_OVERRIDE=0 \
     -DALLOW_PERMISSIVE_SELINUX=0 \
-    -DREBOOT_BOOTLOADER_ON_PANIC=0 \
     -DDUMP_ON_UMOUNT_FAILURE=0
 endif
 
 ifneq (,$(filter eng,$(TARGET_BUILD_VARIANT)))
 init_options += \
-    -DSHUTDOWN_ZERO_TIMEOUT=1
+    -DSHUTDOWN_ZERO_TIMEOUT=1 \
+    -DREBOOT_BOOTLOADER_ON_PANIC=1
 else
 init_options += \
-    -DSHUTDOWN_ZERO_TIMEOUT=0
+    -DSHUTDOWN_ZERO_TIMEOUT=0 \
+    -DREBOOT_BOOTLOADER_ON_PANIC=0
 endif
 
 init_options += -DLOG_UEVENTS=0
-- 
2.7.4

